querydsl-collections/src/main/java/com/querydsl/collections/CollQuery.java

```diff
@@ -74,16 +73,19 @@ public CollQuery(QueryMetadata metadata, QueryEngine queryEngine) {
      * Clone the state of this query to a new CollQuery instance
      */
     @Override
-    public CollQuery clone() {
-        return new CollQuery(queryMixin.getMetadata().clone(), getQueryEngine());
+    public CollQuery<T> clone() {
+        return new CollQuery<T>(queryMixin.getMetadata().clone(), getQueryEngine());
     }
 
-    /**
-     * @return
-     */
     @Override
-    public QueryMetadata getMetadata() {
-        return queryMixin.getMetadata();
+    public <E> CollQuery<E> select(Expression<E> expr) {
+        queryMixin.setProjection(expr);
+        return (CollQuery<E>)this;
```

Comment:
I am not sure. QueryMixin has the query type as an argument, but the select methods change it, so I guess there will be anyway a cast.